src/heartbeat/heartbeat.c

```diff
@@ -78,6 +81,16 @@ volatile bool hb_send_obc_resp = false;
 volatile bool hb_send_eps_resp = false;
 volatile bool hb_send_pay_resp = false;
 
+// Current SSM -> Request restart data from other SSM
+volatile bool hb_received_obc_rdata_req = false;
+volatile bool hb_received_eps_rdata_req = false;
+volatile bool hb_received_pay_rdata_req = false;
+
+// Other SSM -> Send restart data from current SSM
+volatile bool hb_send_obc_rdata_resp = false;
+volatile bool hb_send_eps_rdata_resp = false;
+volatile bool hb_send_pay_rdata_resp = false;
+
```

Comment:
These are necessary, but you will also need to define some new function that will initiate a new request to another subsystem (specified as an argument) for its restart reason and count information.